lib/User.php

```diff
@@ -229,4 +243,81 @@ public function path() {
 	public function slug() {
 		return $this->user_nicename;
 	}
+
+	/**
+	 * Create the array with user roles names and slugs.
+	 *
+	 * @api
+	 * @since 1.8.5
+	 *
+	 * @return array|null
+	 */
+	public function get_roles( $roles ) {
+		if ( empty( $roles ) ) {
+			return null;
+		}
+
+		$wp_roles = wp_roles();
+		$names    = $wp_roles->get_names();
+
+		$values = array();
+
+		foreach( $roles as $role ) {
+			$name = $role;
+			if ( isset( $names[ $role ] ) ) {
+				$name = translate_user_role( $names[ $role ] );
+			}
+
+			$values[$role] = $name;
+		}
+		
+		return $values;
+	}
+
+	/**
+	 * Gets the user roles.
+	 *
+	 * @api
+	 * @since 1.8.5
+	 * @example
+	 * ```twig
+	 * <h2>Role name</h2>
+	 * {% for role in post.author.roles %}
+	 *   {{ role }} 
+	 * {% endfor %}
+	 * ```
+	 * ```twig
+	 * <h2>Role name</h2>
+	 * {{ post.author.roles|join(', ') }}
+	 * ```
+	 * ```twig
+	 * {% for slug in post.author.roles|keys %}
+	 *   {{ slug }} 
+	 * {% endfor %}
+	 * ```
+	 *
+	 * @return array|null
+	 */
+	public function roles() {		
+		return $this->roles;
+	}
+
+	/**
+	 * Checks whether a user has a capability.
+	 *
```

Comment:
```suggestion
	 *
	 * Don’t use role slugs for capability checks. While checking against a role in place of a
	 * capability is supported in part, this practice is discouraged as it may produce unreliable
	 * results. This includes cases where you want to check whether a user is registered. If you
	 * want to check whether a user is a Subscriber, use `{{ user.can('read') }}`. If you only want
	 * to check whether a user is logged in, you can use `{% if user %}`.
	 *
```